include/Gaffer/TweakPlug.inl

```diff
@@ -179,4 +181,95 @@ bool TweaksPlug::applyTweaks(
 	return tweakApplied;
 }
 
+template<typename T>
+T vectorAwareMin( const T &v1, const T &v2 )
+{
+	if constexpr( IECore::TypeTraits::IsVec<T>::value || IECore::TypeTraits::IsColor<T>::value )
+	{
+		T result;
+		for( size_t i = 0; i < T::dimensions(); ++i )
+		{
+			result[i] = std::min( v1[i], v2[i] );
+		}
+		return result;
+	}
+	else
+	{
+		return std::min( v1, v2 );
+	}
+}
+
+template<typename T>
+T vectorAwareMax( const T &v1, const T &v2 )
+{
+	if constexpr( IECore::TypeTraits::IsVec<T>::value || IECore::TypeTraits::IsColor<T>::value )
+	{
+		T result;
+		for( size_t i = 0; i < T::dimensions(); ++i )
+		{
+			result[i] = std::max( v1[i], v2[i] );
+		}
+		return result;
+	}
+	else
+	{
+		return std::max( v1, v2 );
+	}
+}
+
+template< typename T >
+void TweakPlug::applyNumericTweakValue(
+	const T &source,
+	const T &tweak,
+	T &dest,
+	TweakPlug::Mode mode,
+	const std::string &tweakName
+)
+{
+	if constexpr(
+		( std::is_arithmetic_v<T> && !std::is_same_v< T, bool > ) ||
+		IECore::TypeTraits::IsVec<T>::value ||
+		IECore::TypeTraits::IsColor<T>::value
+	 ) {
+		switch( mode )
+		{
+			case TweakPlug::Add :
+				dest = source + tweak;
+				break;
+			case TweakPlug::Subtract :
+				dest = source - tweak;
+				break;
+			case TweakPlug::Multiply :
+				dest = source * tweak;
+				break;
+			case TweakPlug::Min :
+				dest = vectorAwareMin( source, tweak );
+				break;
+			case TweakPlug::Max :
+				dest = vectorAwareMax( source,  tweak );
+				break;
+			case TweakPlug::ListAppend :
+			case TweakPlug::ListPrepend :
+			case TweakPlug::ListRemove :
+			case TweakPlug::Replace :
+			case TweakPlug::Remove :
+			case TweakPlug::Create :
+			case TweakPlug::CreateIfMissing :
+				// These cases are unused - we handle them outside of numericTweak.
+				// But the compiler gets unhappy if we don't handle some cases.
+				assert( false );
+				break;
```

Comment:
I think this warrants an exception - now its in the public API, we can't make the same guarantee.

Reply:
Addressed.